src/daemon/emitter.cc

```diff
@@ -579,6 +608,27 @@ void Emitter::DoRemoteExecute(const base::WorkerPool& pool, ResolveFn resolver,
     failed_tasks_->Push(std::move(*task));
     counter.ReportOnDestroy(true);
   }
+
+  auto new_conf = this->conf();
+
+  // In case if pool was shut down to apply new configuration check if
+  // new number of shards is smaller than our shard. If so, take out tasks
+  // from our shard and distribute them across other shards.
+  if (!all_tasks_->IsClosed() &&
+      new_conf->emitter().shard_queue_limit() > 0 &&
```

Comment:
I've added this to check if new configuration supports/expects sharding.
There might be cases with updates of configuration without shards to another configuration without shards.
In this case there's nothing to be done.